straxen/plugins/events/corrected_areas.py

```diff
@@ -297,51 +313,106 @@ def apply_s2_corrections(
+            cs2_w_xy_elife: S2xy, elife
+            cs2_area_fraction_top_w_xy_elife: S2xy, elife
+
+        The values for peak bias, SEG/EE and elife have no effect on the AFT.
+        They are still included in this docstring to highlight which actual
+        values are used to compute the AFT. This way, if there is a difference
+        in AFT with vs without elife we know there is a bug in this funtcion.
 
         Returns:
             cs2,
             cs2_area_fraction_top,
-            cs2_before_pi,
-            cs2_area_fraction_top_before_pi,
-            cs2_before_elife,
-            cs2_area_fraction_top_before_elife,
-            cs2_wo_segee_picorr,
-            cs2_area_fraction_top_wo_segee_picorr,
+            cs2_w_bias,
+            cs2_area_fraction_top_w_bias,
+            cs2_w_bias_xy,
+            cs2_area_fraction_top_w_bias_xy,
+            cs2_w_bias_segee,
+            cs2_area_fraction_top_w_bias_segee,
+            cs2_w_bias_xy_segee,
+            cs2_area_fraction_top_w_bias_xy_segee,
+            cs2_w_bias_xy_elife,
+            cs2_area_fraction_top_w_bias_xy_elife,
+            cs2_w_elife,
+            cs2_area_fraction_top_w_elife,
+            cs2_w_xy_elife,
+            cs2_area_fraction_top_w_xy_elife,
 
         """
         # Base areas
         s2_area_top = s2_area * s2_aft
         s2_area_bottom = s2_area * (1 - s2_aft)
 
-        # Fully corrected S2 (in stages for intermediate variables)
-        cs2_top_before_pi = s2_area_top / s2_bias_correction / s2_xy_correction_top / seg_ee_corr
-        cs2_bottom_before_pi = (
-            s2_area_bottom / s2_bias_correction / s2_xy_correction_bottom / seg_ee_corr
-        )
-        cs2_before_pi = cs2_top_before_pi + cs2_bottom_before_pi
-
-        cs2_top_before_elife = cs2_top_before_pi
-        cs2_bottom_before_elife = cs2_bottom_before_pi * pi_corr_bottom
-        cs2_before_elife = cs2_top_before_elife + cs2_bottom_before_elife
+        # First, we apply all corrections in the "normal" order up to cs2
+        # peak bias
+        cs2_top_w_bias = s2_area_top / s2_bias_correction
+        cs2_bottom_w_bias = s2_area_bottom / s2_bias_correction
+        cs2_w_bias = cs2_top_w_bias + cs2_bottom_w_bias
+        # S2 xy
+        cs2_top_w_bias_xy = cs2_top_w_bias / s2_xy_correction_top
+        cs2_bottom_w_bias_xy = cs2_bottom_w_bias / s2_xy_correction_bottom
+        cs2_w_bias_xy = cs2_top_w_bias_xy + cs2_bottom_w_bias_xy
+        # SEG/EE
+        cs2_top_w_bias_xy_segee = cs2_top_w_bias_xy / seg_ee_corr
+        cs2_bottom_w_bias_xy_segee = cs2_bottom_w_bias_xy / seg_ee_corr
+        cs2_w_bias_xy_segee = cs2_top_w_bias_xy_segee + cs2_bottom_w_bias_xy_segee
+        # elife
+        cs2_top = cs2_top_w_bias_xy_segee * elife_correction
+        cs2_bottom = cs2_bottom_w_bias_xy_segee * elife_correction
+        cs2 = cs2_top + cs2_bottom
+        # PI AFT
+        cs2_area_fraction_top = cs2_top / (cs2_top + cs2_bottom * pi_corr_bottom)
```

Comment:
There might be a very common bug. For S2-Only events, the `elife_correction` will be `np.nan`, so `cs2_area_fraction_top` will be `np.nan` if `cs2_top` is affected by `elife_correction`.

Reply:
Please make sure also in the docstring that the elife does not affect aft.

Reply:
Sorry I was wrong, this line https://github.com/XENONnT/straxen/blob/e9fe5dd7fb61fb537d35732730b41639ebc39980/straxen/plugins/events/corrected_areas.py#L532 guaranteed that the S2-Only events' AFTs are not np.nan

Reply:
It should be `cs2_area_fraction_top_wo_elifecorr`, but later we assigned `cs2_area_fraction_top` as `cs2_area_fraction_top_wo_elifecorr` so in cutax code it is `cs2_area_fraction_top` https://github.com/XENONnT/cutax/blob/1e2d98cfde1683c52c873c5eaad63f260b93003a/cutax/cuts/s2_aft.py#L93